src/bin/pgcopydb/ld_transform.c

```diff
@@ -1169,6 +1169,64 @@ parseMessage(StreamContext *privateContext, char *message, JSON_Value *json)
 }
 
 
+/* append the new insert statement if targets same table as previous */
+static bool
+coalesceInsertStatements(LogicalMessageInsert *last,
+						 LogicalMessageInsert *stmt)
+{
+	if (strcmp(last->nspname, stmt->nspname) != 0 ||
+		strcmp(last->relname, stmt->relname) != 0)
+	{
+		return false;
+	}
+
+	LogicalMessageValuesArray *lastValuesArray = &last->new.array->values;
+
+	if (lastValuesArray->capacity < (lastValuesArray->count + 1))
+	{
+		/* double the capacity to avoid excessive realloc */
+		lastValuesArray->capacity *= 2;
+		lastValuesArray->array = (LogicalMessageValues *) realloc(lastValuesArray->array,
+																  sizeof(
+																	  LogicalMessageValues)
+																  *
+																  lastValuesArray->
+																  capacity);
```

Comment:
What if realloc fails? Please check for a NULL pointer being returned. BTW that's when we return false.

Nitpicking: review the formatting, it's too far on the right here.